rift/rib.py

```diff
@@ -104,71 +187,111 @@ def nr_destinations(self):
 
     def nr_routes(self):
         count = 0
-        for destination in self.destinations.values():
+        for prefix in self.destinations:
+            destination = self.destinations.get(prefix)
             count += len(destination.routes)
         return count
 
-class _Destination:
 
-    # For each prefix, there can be up to one route per owner. This is also the order of preference
-    # for the routes from different owners to the same destination (higher numerical value is more
-    # preferred)
+class _Destination:
+    """
+    Class that contains all routes for a given prefix destination
+    Attributes of this class are:
+        - rib: reference to the RIB
+        - prefix: prefix associated to this destination
+        - routes: list of RibRoute objects, in decreasing order or owner (= in decreasing order of
+            preference, higher numerical value is more preferred).
+            For a given owner, at most one route is allowed to be in the list
+    """
 
-    def __init__(self, prefix):
+    def __init__(self, rib, prefix):
+        self.rib = rib
         self.prefix = prefix
-        # List of Route objects, in decreasing order or owner (= in decreasing order of preference)
-        # For a given owner, at most one route is allowed to be in the list
         self.routes = []
 
+    @property
+    def parent_prefix_dest(self):
+        """
+        :return: the Destination object associated to the parent prefix of the current one
+        """
+        parent_prefix = self.rib.destinations.parent(packet_common.ip_prefix_str(self.prefix))
+        if parent_prefix is None:
+            return None
+
+        return self.rib.destinations.get(parent_prefix)
+
+    @property
+    def best_route(self):
+        """
+        :return: the best RibRoute for this prefix
+        """
+        return self.routes[0]
+
     def get_route(self, owner):
+        """
+        Get RibRoute object for a given owner if present
+        :param owner: (int) owner of the route
+        :return: (RibRoute|None) desired RibRoute object if present, else None
+        """
         for rte in self.routes:
             if rte.owner == owner:
                 return rte
         return None
 
-    def update_fib(self, fib):
```

Comment:
What did you not like about the old way of updating the FIB?

Reply:
The old method had a problem with responsibility assignment. We redesigned the Destination class a little bit. We thought that it should only contain information about the routes for a given destination and it should not have the responsibility to access the FIB (which is a RIB responsibility)